src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,225 @@
+            }
+            pred = newNode;
+        }
+
+        if (succ == null) {
+            tail = pred;
+        } else {
+            pred.next = succ;
+            succ.prev = pred;
+        }
+        size += numNew;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return node(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> current = node(index);
+        T finVal = current.value;
+        current.value = value;
+        return finVal;
+    }
+
+    public Node<T> node(int index) {
+        if (index < (size >> 1)) {
+            Node<T> x = head;
+            for (int i = 0; i < index; i++) {
+                x = x.next;
+            }
+            return x;
+        } else {
+            Node<T> x = tail;
+            for (int i = size - 1; i > index; i--) {
+                x = x.prev;
+            }
+            return x;
+        }
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        if (index == 0) {
+            final T temp = head.value;
+            head = head.next;
+            size--;
+            if (head == null) {
+                tail = null;
+            }
+            return temp;
+        } else if (index == size - 1) {
+            if (size == 0) {
+                return null;
+            } else if (size == 1) {
+                T temp = head.value;
+                head = tail = null;
+                size = 0;
+                return temp;
+            } else {
+                Node<T> current = head;
+                for (int i = 0; i < size - 2; i++) {
+                    current = current.next;
+                }
+                final T temp = tail.value;
+                tail = current;
+                tail.next = null;
+                size--;
+                return temp;
+            }
+        } else {
+            Node<T> previous = head;
+            for (int i = 0; i < index; i++) {
+                previous = previous.next;
+            }
+            Node<T> current = previous.next;
+            previous.next = current;
+            current.prev = previous.prev;
+            previous.prev.next = current;
+            T temp = previous.value;
+            size--;
+            return temp;
+        }
     }
 
     @Override
     public boolean remove(T object) {
+        if (object == null) {
+            for (Node<T> x = head; x != null; x = x.next) {
+                if (x.value == null) {
+                    unlink(x);
+                    return true;
+                }
+            }
+        } else {
+            for (Node<T> x = head; x != null; x = x.next) {
+                if (object.equals(x.value)) {
+                    unlink(x);
+                    return true;
+                }
+            }
+        }
         return false;
```

Comment:
1. please don't use 1-letter variable naming except obvious counter/index indicator
2. Two cycles may be combined:
```suggestion
        for (Node<T> currentNode = head; currentNode != null; currentNode = currentNode.next) {
            if (currentNode.value == object || 
                    currentNode.value != null && currentNode.value.equals(object)) {
                unlink(currentNode);
                return true;
            }
        }
        return false;
```